pkg/controllers/sync/controller.go

```diff
@@ -717,88 +719,76 @@ func (s *SyncController) setFederatedStatus(fedResource FederatedResource, colli
 		return false, errors.Wrapf(err, "failed to update resource")
 	})
 	if err != nil {
-		runtime.HandleError(errors.Wrapf(err, "failed to update syncing annotation for %s %q", kind, name))
+		kindedKeyedLogger.Error(err, "Failed to update syncing annotation")
 		return worker.StatusError
 	}
 
 	return worker.StatusAllOK
 }
 
-func (s *SyncController) ensureDeletion(fedResource FederatedResource) worker.Result {
+func (s *SyncController) ensureDeletion(ctx context.Context, fedResource FederatedResource) worker.Result {
 	fedResource.DeleteVersions()
 
 	key := fedResource.FederatedName().String()
 	kind := fedResource.FederatedKind()
+	kindedKeyedLogger := klog.FromContext(ctx).WithValues("federated-kind", kind)
```

Comment:
`kindedKeyedLogger` seems unnecessarily verbose, can we replace its usages with `keyedLogger` or even just `logger`?

Reply:
done, pls take review.